src/appengine/handlers/login.py

```diff
@@ -81,5 +84,6 @@ def get(self):
       # Even if the revoke failed, remove the cookie.
       logs.log_error('Failed to revoke session cookie.')
 
-    self.response.delete_cookie('session')
-    self.redirect(self.request.get('dest'))
+    response = Response()
+    response.delete_cookie('session')
+    return self.redirect(request.get('dest'), response=response)
```

Comment:
I'm a bit confused how this works here. redirect doesn't take in a response argument? 

And again, we can delete the cookie from after getting the return value from self.redirect instead of doiung this.